servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```
@@ -2664,6 +2665,40 @@ public final Publisher<T> skipWhile(Predicate<? super T> predicate) {
         return filter(FilterPublisher.skipWhileSupplier(predicate));
     }
 
+    /**
+     * Converts this {@link Publisher} to a {@link Single} that will contain the first element of this {@link Publisher}
+     * and a {@link Publisher} representing the remainder of the stream, to a mapping function.
+     * <p>
+     * Note that either the packer function itself or any operator following this one MUST eventually take care of the
+     * tail {@link Publisher} or they risk leaking resources. This includes cases where an exception is thrown before
+     * returning a result. In addition, the tail may only be subscribed to exactly one time. For instance, using memory
```

Consider: where an exception is thrown instead of returning the packed object or a failed Single is returned.